craft_providers/lxd/lxc.py

```diff
@@ -793,10 +793,19 @@ def project_create(self, *, project: str, remote: str = "local") -> None:
         try:
             self._run_lxc(command, capture_output=True, project=project)
         except subprocess.CalledProcessError as error:
-            raise LXDError(
-                brief=f"Failed to create project {project!r}.",
-                details=errors.details_from_called_process_error(error),
-            ) from error
+            # handle the race condition where two processes check and
+            # create the same project at the same time
+            if project in self.project_list(remote=remote):
```

Comment:
maybe instead of doing a check in exception, we can do a check `if project in self.project_list` before we try to run ` self._run_lxc` or this will not work?

Reply:
This will still have race chance. 2 processes see the project not exists and try create it together. Since the only lock is in the LXC.

Reply:
Sheng is right - the lock is in LXC.  Craft-providers already checks if the project exists before calling this function, but it occurs in a different module ([here](https://github.com/canonical/craft-providers/blob/bb4ff5a849cb81199b5630099080dbbb0499cecf/craft_providers/lxd/launcher.py#L125-L130))

Reply:
I am not 100% sure. How deterministic is this approach? 

can we end up that the error will be again false positive due to race?
will LXD return an error if "the project is in creation process" and we still call "create project" ?

Reply:
Create project should be close to atomic and very fast.

Reply:
Ah yes, that's a fair point.  I'm not sure if LXC will block or fail on `lxc project list` if `lxc project create` is in process.

This approach worked for `lxc remote list` and `lxc remote add`.  

If this race condition continues to occur after this PR or we get a report of another command failing (rule of 3), then I think we will look into other solutions, such as creating a utility function with similar logic plus some retry/backoff timing.

Reply:
agreed